registry/api/errcode/errors.go

```diff
@@ -72,24 +76,24 @@ func (ec *ErrorCode) UnmarshalText(text []byte) error {
 // set the Detail property appropriately
 func (ec ErrorCode) WithDetail(detail interface{}) Error {
 	return Error{
-		Code:    ec,
-		Message: ec.Message(),
+		Code: ec,
+		Msg:  ec.Message(),
 	}.WithDetail(detail)
 }
 
 // WithArgs creates a new Error struct and sets the Args slice
 func (ec ErrorCode) WithArgs(args ...interface{}) Error {
 	return Error{
-		Code:    ec,
-		Message: ec.Message(),
+		Code: ec,
+		Msg:  ec.Message(),
 	}.WithArgs(args...)
 }
 
 // Error provides a wrapper around ErrorCode with extra Details provided.
 type Error struct {
-	Code    ErrorCode   `json:"code"`
-	Message string      `json:"message"`
-	Detail  interface{} `json:"detail,omitempty"`
+	Code   ErrorCode   `json:"code"`
+	Msg    string      `json:"message"`
```

Comment:
At some point in the future its very possible that the message will be dynamically generated and I'd prefer to future proof how people access it. Yes right now its a static string but if/when we get around to making it dynamic I don't want all uses of err.Message to have to change to err.Message() - we should do this now before we have too many users.